<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Pipeline:4-RNA-seq and a minimal RNA-seq cook book</title>
  <meta name="description" content="This is a minimal example of how to process RNA-seq">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Pipeline:4-RNA-seq and a minimal RNA-seq cook book" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of how to process RNA-seq" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Pipeline:4-RNA-seq and a minimal RNA-seq cook book" />
  
  <meta name="twitter:description" content="This is a minimal example of how to process RNA-seq" />
  

<meta name="author" content="Rishi Das Roy, Postdoctoral Fellow, Evo-Devo lab, University of Helsinki">


<meta name="date" content="2019-05-23">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  

<link rel="next" href="quality-control.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#structure-of-the-book"><i class="fa fa-check"></i>Structure of the book</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#chapter-1-quality-control"><i class="fa fa-check"></i>Chapter 1: Quality Control</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#chapter-2-alignment-and-count"><i class="fa fa-check"></i>Chapter 2: Alignment and count</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#chapter-3-read-count-analysis"><i class="fa fa-check"></i>Chapter 3: Read count analysis</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#chapter-4-pipeline4-rna-seq"><i class="fa fa-check"></i>Chapter 4: Pipeline:4-RNA-seq</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgement"><i class="fa fa-check"></i>Acknowledgement</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>1</b> Quality control</a><ul>
<li class="chapter" data-level="1.1" data-path="quality-control.html"><a href="quality-control.html#getting-your-share-of-the-data"><i class="fa fa-check"></i><b>1.1</b> Getting your share of the data</a></li>
<li class="chapter" data-level="1.2" data-path="quality-control.html"><a href="quality-control.html#workflow"><i class="fa fa-check"></i><b>1.2</b> Workflow</a></li>
<li class="chapter" data-level="1.3" data-path="quality-control.html"><a href="quality-control.html#step-1.-qa-of-raw-reads"><i class="fa fa-check"></i><b>1.3</b> Step 1. QA of raw reads</a><ul>
<li class="chapter" data-level="1.3.1" data-path="quality-control.html"><a href="quality-control.html#fastqc-on-rest-of-the-fastq-files"><i class="fa fa-check"></i><b>1.3.1</b> FastQC on rest of the fastq files</a></li>
<li class="chapter" data-level="1.3.2" data-path="quality-control.html"><a href="quality-control.html#summerize"><i class="fa fa-check"></i><b>1.3.2</b> Summerize</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="quality-control.html"><a href="quality-control.html#step-2.-filtering-rrna"><i class="fa fa-check"></i><b>1.4</b> Step 2. Filtering rRNA</a></li>
<li class="chapter" data-level="1.5" data-path="quality-control.html"><a href="quality-control.html#step-3.-quality-trimming-and-adapter-removal"><i class="fa fa-check"></i><b>1.5</b> Step 3. Quality trimming and adapter removal</a><ul>
<li class="chapter" data-level="1.5.1" data-path="quality-control.html"><a href="quality-control.html#trimmed-data-fastqc"><i class="fa fa-check"></i><b>1.5.1</b> Trimmed data FastQC</a></li>
<li class="chapter" data-level="1.5.2" data-path="quality-control.html"><a href="quality-control.html#final-filtering-report"><i class="fa fa-check"></i><b>1.5.2</b> Final filtering report</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="alignment-and-count.html"><a href="alignment-and-count.html"><i class="fa fa-check"></i><b>2</b> Alignment and count</a><ul>
<li class="chapter" data-level="2.1" data-path="alignment-and-count.html"><a href="alignment-and-count.html#step-4.-alignment-of-reads"><i class="fa fa-check"></i><b>2.1</b> Step 4. Alignment of reads</a><ul>
<li class="chapter" data-level="2.1.1" data-path="alignment-and-count.html"><a href="alignment-and-count.html#alignment-to-the-genome"><i class="fa fa-check"></i><b>2.1.1</b> Alignment to the genome</a></li>
<li class="chapter" data-level="2.1.2" data-path="alignment-and-count.html"><a href="alignment-and-count.html#now-align-the-reads"><i class="fa fa-check"></i><b>2.1.2</b> Now align the reads</a></li>
<li class="chapter" data-level="2.1.3" data-path="alignment-and-count.html"><a href="alignment-and-count.html#post-processing-the-alignment-result-files"><i class="fa fa-check"></i><b>2.1.3</b> Post-processing the alignment result files</a></li>
<li class="chapter" data-level="2.1.4" data-path="alignment-and-count.html"><a href="alignment-and-count.html#visualization"><i class="fa fa-check"></i><b>2.1.4</b> Visualization</a></li>
<li class="chapter" data-level="2.1.5" data-path="alignment-and-count.html"><a href="alignment-and-count.html#quality-check-of-alignment"><i class="fa fa-check"></i><b>2.1.5</b> Quality check of alignment</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="alignment-and-count.html"><a href="alignment-and-count.html#step-5.-read-count"><i class="fa fa-check"></i><b>2.2</b> Step 5. Read count</a><ul>
<li class="chapter" data-level="2.2.1" data-path="alignment-and-count.html"><a href="alignment-and-count.html#counting-reads-per-feature"><i class="fa fa-check"></i><b>2.2.1</b> Counting reads per feature</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="alignment-and-count.html"><a href="alignment-and-count.html#step-6.-final-report"><i class="fa fa-check"></i><b>2.3</b> Step 6. Final report</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="read-count-analysis.html"><a href="read-count-analysis.html"><i class="fa fa-check"></i><b>3</b> Read count analysis</a><ul>
<li class="chapter" data-level="3.0.1" data-path="read-count-analysis.html"><a href="read-count-analysis.html#data-from-taito"><i class="fa fa-check"></i><b>3.0.1</b> Data from Taito</a></li>
<li class="chapter" data-level="3.0.2" data-path="read-count-analysis.html"><a href="read-count-analysis.html#reading-the-metadata"><i class="fa fa-check"></i><b>3.0.2</b> Reading the metadata</a></li>
<li class="chapter" data-level="3.0.3" data-path="read-count-analysis.html"><a href="read-count-analysis.html#summarizing-an-rna-seq-experiment-as-a-count-matrix"><i class="fa fa-check"></i><b>3.0.3</b> Summarizing an RNA-seq experiment as a count matrix</a></li>
<li class="chapter" data-level="3.0.4" data-path="read-count-analysis.html"><a href="read-count-analysis.html#installation"><i class="fa fa-check"></i><b>3.0.4</b> Installation</a></li>
<li class="chapter" data-level="3.0.5" data-path="read-count-analysis.html"><a href="read-count-analysis.html#htseq-count"><i class="fa fa-check"></i><b>3.0.5</b> htseq-count</a></li>
<li class="chapter" data-level="3.1" data-path="read-count-analysis.html"><a href="read-count-analysis.html#qc"><i class="fa fa-check"></i><b>3.1</b> QC</a><ul>
<li class="chapter" data-level="3.1.1" data-path="read-count-analysis.html"><a href="read-count-analysis.html#expression"><i class="fa fa-check"></i><b>3.1.1</b> Expression</a></li>
<li class="chapter" data-level="3.1.2" data-path="read-count-analysis.html"><a href="read-count-analysis.html#visual-exploration"><i class="fa fa-check"></i><b>3.1.2</b> Visual Exploration</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="read-count-analysis.html"><a href="read-count-analysis.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.2</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="3.2.1" data-path="read-count-analysis.html"><a href="read-count-analysis.html#diagnostic-plots"><i class="fa fa-check"></i><b>3.2.1</b> Diagnostic plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html"><i class="fa fa-check"></i><b>4</b> Pipeline 4-RNA-seq</a><ul>
<li class="chapter" data-level="4.0.1" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#dependency"><i class="fa fa-check"></i><b>4.0.1</b> Dependency</a></li>
<li class="chapter" data-level="4.0.2" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#installation-1"><i class="fa fa-check"></i><b>4.0.2</b> Installation</a></li>
<li class="chapter" data-level="4.1" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#how-to-run"><i class="fa fa-check"></i><b>4.1</b> How to run</a><ul>
<li class="chapter" data-level="4.1.1" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#download-public-resources"><i class="fa fa-check"></i><b>4.1.1</b> Download public resources</a></li>
<li class="chapter" data-level="4.1.2" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#step-1-3.-qc-and-filtering"><i class="fa fa-check"></i><b>4.1.2</b> Step 1-3. QC and Filtering</a></li>
<li class="chapter" data-level="4.1.3" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#step-4.-alignment-of-reads-1"><i class="fa fa-check"></i><b>4.1.3</b> Step 4. Alignment of reads</a></li>
<li class="chapter" data-level="4.1.4" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#step-5.-read-count-1"><i class="fa fa-check"></i><b>4.1.4</b> Step 5. Read count</a></li>
<li class="chapter" data-level="4.1.5" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#final-report"><i class="fa fa-check"></i><b>4.1.5</b> Final report</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#extra"><i class="fa fa-check"></i><b>4.2</b> EXTRA</a><ul>
<li class="chapter" data-level="4.2.1" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#alignment-read-viewer"><i class="fa fa-check"></i><b>4.2.1</b> Alignment read viewer</a></li>
<li class="chapter" data-level="4.2.2" data-path="pipeline-4-rna-seq.html"><a href="pipeline-4-rna-seq.html#compressing-fastq-files"><i class="fa fa-check"></i><b>4.2.2</b> Compressing fastq files</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pipeline:4-RNA-seq and a minimal RNA-seq cook book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Pipeline:4-RNA-seq and a minimal RNA-seq cook book</h1>
<p class="author"><em>Rishi Das Roy, Postdoctoral Fellow, Evo-Devo lab, University of Helsinki</em></p>
<p class="date"><em>2019-05-23</em></p>
</div>
<div id="preface" class="section level1 unnumbered">
<h1>Preface</h1>
<p>The high throughput RNA-sequencing technology has been widely used and has advanced our understanding of RNA biology. The process of RNA-sequencing can be divided in two parts; 1) Generating the data from biological sample and 2) Computational data analysis. This book is about the second part which is usually done with a subset of many related software, refered as pipeline. For ease of biologist, there are many predefined and I have been analyzing RNA-seq data from 2014 using different. This book is an attempt to describe how to use the pipeline <a href="https://github.com/vondoRishi/4-RNA-seq">4-RNA-seq</a> and it is also a collection of the study materials which has been used in RNA-seq section of the course <a href="https://moodle.helsinki.fi/course/view.php?id=31702">Advanced course in genomics and gene regulation 2019</a> by me.</p>
<div id="structure-of-the-book" class="section level2 unnumbered">
<h2>Structure of the book</h2>
<p>Chapter 1 and 2 explains the step wchich are executed from a unix terminal. Chapter 3 describes the final data analysis step which requires R and can be performed in any OS. Chapter 4 explains 4-RNA-seq pipleline which to execute chapter 1 and 2 commands in easier and efficient ways. Therefore chapter 1 to 3 is must for beginner and chapter 4 is for who have slurm environment and looking for a pipeline to execute RNA-seq data.</p>
<div id="chapter-1-quality-control" class="section level3 unnumbered">
<h3>Chapter 1: Quality Control</h3>
<p>Description of quality assessment, filtering ribosomal RNA and bad reads of fastq files with fastqc, multiqc, sortmerna and trimmomatic.</p>
</div>
<div id="chapter-2-alignment-and-count" class="section level3 unnumbered">
<h3>Chapter 2: Alignment and count</h3>
<p>Alighnment of reads and count values from fastq and bam files using toll like star and htseq-count.</p>
</div>
<div id="chapter-3-read-count-analysis" class="section level3 unnumbered">
<h3>Chapter 3: Read count analysis</h3>
<p>Differential expression analysis with R package DESeq2.</p>
</div>
<div id="chapter-4-pipeline4-rna-seq" class="section level3 unnumbered">
<h3>Chapter 4: Pipeline:4-RNA-seq</h3>
<p>A step by step instruction for 4-RNA-seq pipeline execution.</p>
</div>
</div>
<div id="acknowledgement" class="section level2 unnumbered">
<h2>Acknowledgement</h2>
<p>Most of the study material are motivated from <strong><a href="http://www.finbionet.fi/ils/ngg2016/ws_ngg.html">Rna-seq data analysis, Workshop II</a></strong> 1-3.11.2016. This could not be possible without the encouragement and permission from the istructor/author of the workshop. I like to thank<br />
* Nicolas DelHomme, Umeå University, Sweden<br />
* Bastian Schiffthaler, Umeå University, Sweden<br />
* Charlotte Soneson, Friedrich Miescher Institute for Biomedical Research<br />
* Petri Törönen, University of Helsinki for reviewing this book.</p>
<p>The online version of this book is licensed under the <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="quality-control.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
